kong/runloop/handler.lua

```diff
@@ -1046,10 +1046,22 @@ return {
       var.upstream_host = upstream_url_t.host
     end,
     after = function(ctx)
+      local upstream_scheme = var.upstream_scheme
+
+      local balancer_data = ctx.balancer_data
+      balancer_data.scheme = upstream_scheme -- COMPAT: pdk
+
+      local upstream_host = var.upstream_host
```

Comment:
Do we also need to set `preserve_host == true` here?

Reply:
Since this only involves the stream module, I don't think it would be useful to add it?

Reply:
added